src/main/java/org/kohsuke/github/GHRepository.java

```diff
@@ -2961,6 +2968,36 @@ public GHWorkflowRun getWorkflowRun(long id) throws IOException {
                 .wrapUp(this);
     }
 
+    /**
+     * Lists all the artifacts of this repository.
+     *
+     * @return the paged iterable
+     */
+    public PagedIterable<GHArtifact> listArtifacts() {
+        try {
+            GitHubRequest request = root.createRequest().withUrlPath(getApiTailUrl("actions/artifacts")).build();
+            return new GHArtifactsIterable(this, request);
+        } catch (MalformedURLException e) {
+            throw new GHException("Malformed URL", e);
+        }
```

Comment:
Here's that try-catch again.  I figured out why most other `list*` methods don't need this: 

https://github.com/hub4j/github-api/blob/a68d16d5de6da437fa493f3bd524de03eaaee1ad/src/main/java/org/kohsuke/github/Requester.java#L153-L158

This try catch appears in two places so in this case I definitely want it inside the iterable.

Reply:
I knew you would say that but the query is different in both cases, thus why I haven't changed it as for the other PR. So should I pass the builder instead of the request and build the request in the iterable?

Frankly, for me the issue is with the initial design decision of throwing low value unchecked exceptions from nearly everywhere.

Reply:
> I knew you would say that but the query is different in both cases, thus why I haven't changed it as for the other PR. So should I pass the builder instead of the request and build the request in the iterable?

Yeah, pass the builder.  Or maybe the right thing is to make a version of the `build()` that wraps those exceptions and use that in all 3 (4?) calls.  I would change the behavior of `build()` but that could result in client that understand how to handle certain exceptions suddenly not getting those anymore. 

> Frankly, for me the issue is with the initial design decision of throwing low value unchecked exceptions from nearly everywhere.

Absolutely.  Please file an issue and describe how it should work in the next version.  I'm interested.  Some of the current behavior is probably cruft from long removed code.  It may also be a result of the underlying implementation of `HttpConnector` and `HttpUrlConnection` not being controlled by this library.  `okhttp3` has notably different behavior from the Java built-in implementation.

Reply:
OK, for now, I passed the builder to the iterable.

I think more involved changes need to be thought out a bit more. I'll think about it and open an issue.